### DISCUSSION TITLE
HTTP Client Bug

### DISCUSSION THREAD

**Discussion by khobiziilyes** [laravel/framework#33440] [General] [answered]
Hello,
This is my first time for me to open a bug report on Github so excuse me if i did anything wrong and guide me please.
I have bad english too :) as i'm not a native speaker so ...
And thanks for the great framework ! it took me only one week to get familiar with it ! Thanks a lot
I found a small bug the [Laravel HTTP Client ](https://laravel.com/docs/7.x/http-client)
The problem happens when sending multipart post request
Try this code and you will see :+1:
`
use Illuminate\Support\Facades\Http;
$client = Http::attach('anything', '0');
$client->post('localhost');
`
As you can see it says "A 'contents' key is required "
I think that Laravel validate the datas using "If" and because of PHP type casting it converts "0" to false
And that make the data "no set" and that's why that problem happens ...
I hope you can fix this shortly ...
Thanks !

**Comment by jmarcher** [comment, marked as answer]
`Http::attach(...)` is used to attach a file to the `Request` if you really need a file with a `0` you can append the contents with `\0`

**Reply by khobiziilyes** [reply]
The API im using requires multipart/form-data ... So i have to use attach instead of anything else ...

**Reply by jmarcher** [reply]
No, if what you want is to send data, you need to use the second parameter of `->post(...)` which is the data you want to send.
